timeplus-native-jdbc/src/test/java/com/timeplus/jdbc/type/AggregationTypeTest.java

```diff
@@ -0,0 +1,96 @@
+    //     -- (since finalizeChunk() accepts positions and they may be wrong)
+    //     b int64,
+    //     a int64,
+    //     grp_aggreg aggregate_function(group_array_array, array(uint64))
+    // ) engine = MergeTree() order by a;
+    // insert into data_02295 select 0 as b, int_div(number, 2) as a, group_array_array_state([to_uint64(number)]) from numbers(4) group by a, b;
+    // SELECT grp_aggreg FROM data_02295 GROUP BY a, grp_aggreg WITH TOTALS ORDER BY a SETTINGS optimize_aggregation_in_order = 0 FORMAT JSONEachRow;
+    
+    @Test
+    public void testType() throws Exception {
+        withStatement(statement -> {
+            statement.execute("DROP STREAM IF EXISTS agg_test");
+            statement.execute(
+                    "CREATE STREAM agg_test (" +
+                    "b int64,"+
+                    "a int64, "+ 
+                    "grp_aggreg aggregate_function(group_array_array, array(uint64))"+
+                    ") engine = MergeTree() order by a;");
+
+            statement.executeQuery("INSERT INTO agg_test select 0 as b, int_div(number, 2) as a, group_array_array_state([to_uint64(number)]) from numbers(4) group by a, b;");
+
+            ResultSet rs = statement.executeQuery("SELECT grp_aggreg FROM agg_test GROUP BY a, grp_aggreg WITH TOTALS ORDER BY a SETTINGS optimize_aggregation_in_order = 0;");
+            int size = 0;
+            while (rs.next()) {
+                size++;
+                Object obj = rs.getObject(1);
+                assertNotNull(obj);
+            }
+            assertEquals(2, size);
+
+            rs = statement.executeQuery("SELECT a, min(b), max(b) FROM agg_test GROUP BY a ORDER BY a, count() SETTINGS optimize_aggregation_in_order = 1, max_threads = 1;");
+            size = 0;
+            while (rs.next()) {
+                size++;
+                Object obj = rs.getObject(1);
+                assertNotNull(obj);
+            }
+            assertEquals(2, size);
+            statement.execute("DROP STREAM IF EXISTS agg_test");
+        });
+    }
+
+    @Test
+    public void testComplexType() throws Exception {
+        withStatement(statement -> {
+            statement.execute("DROP STREAM IF EXISTS agg_test");
+            statement.execute(
+                    "CREATE STREAM agg_test (" +
+                    "a int64, grp_aggreg aggregate_function(group_array_array, array(uint64)),"+
+                    "grp_simple simple_aggregate_function(group_array_array, array(uint64))) engine = MergeTree() order by a");
+
+            statement.executeQuery("INSERT INTO agg_test select 1 as a, group_array_array_state([to_uint64(number)]), group_array_array([to_uint64(number)]) from numbers(2) group by a;");
+
+            ResultSet rs = statement.executeQuery("SELECT array_sort(group_array_array_merge(grp_aggreg)) as gra , array_sort(group_array_array(grp_simple)) as grs FROM agg_test group by a SETTINGS optimize_aggregation_in_order=1;");
```

Comment:
we need cover more cases:
1) select <type>: to read intermediate result
2) select aggr(type): to read final aggregated result